### PR TITLE
Added a new feature to add an offset for the bubble.

### PR DESCRIPTION
**Pull request by JaimeYesidLeonParada**
Add bubble offset.

### CODE REVIEW

**Comment by JaimeYesidLeonParada** [comment]
@andreamazz How do you see the PR? I am waiting for your comments. Thanks

**Comment by andreamazz** [comment]
Hi @JaimeYesidLeonParada 
Thanks for the PR.

I think it would be a good idea to add a check on the bubble offset's value, values too large can break the layout. If the value is higher than the available space, the excess should be ignored. So if I set an offset of 100 while having only 25 points of useful space, 25 should be the value used while building the popover.  

It would also be great if you can add a test in the test project to make sure that future release won't break your changes 👍

**Comment by JaimeYesidLeonParada** [comment]
@andreamazz ok, no problem.

**Comment by JaimeYesidLeonParada** [comment]
Hi @andreamazz the PR was updated following your comments, I will appreciate your check. Thanks.

File: PopTipDemo/PopTipDemoTests/PopTipDemoTests.m [hunk 1]
```diff
@@ -146,6 +145,21 @@ - (void)pulseAnimation;
             expect([subject valueForKey:@"maxWidth"]).to.equal(200);
         });
     });
+    
+    describe(@"showText:direction:maxWidth:inView:fromFrame:", ^{
```
Comment (andreamazz) [inline PopTipDemo/PopTipDemoTests/PopTipDemoTests.m]: Why add a repeated test? You can add the new expectation in the previous one.

**Comment by JaimeYesidLeonParada** [comment]
@andreamazz you are right, I deleted the duplicated test.

**Comment by JaimeYesidLeonParada** [comment]
Hi @andreamazz  I hope that the implementation is complete and you can approve the PR. Thanks

**Comment by andreamazz** [comment]
It's still not quite right, in the demo project setting an offset of `100` causes this:
http://d.pr/i/1291h

**Comment by JaimeYesidLeonParada** [comment]
It is right @andreamazz , because the programer handle the offset; in the last commit I limited the bubble to the frame of the view where is showed.
Maybe do you want that the offset bubble also will be limited  to the arrow position?

**Comment by andreamazz** [comment]
I see this more like an offset of the arrow itself. I like the API to be as less error prone as possible, so the arrow should always be there, anchoring the bubble. If you need to push the bubble further away, you should present it from a bigger frame.
So, if the offset is too high, it should be limited so that the arrow is anchoring it to its limit, like this:
pop

**Comment by JaimeYesidLeonParada** [comment]
@andreamazz sound good your approach, I am working on it.

**Comment by JaimeYesidLeonParada** [comment]
@andreamazz PR updated following comments.

**Comment by andreamazz** [comment]
Ace! Merged.
Thanks for your work

**Pull request merged**